apis/dns/v1alpha1/managed_zone_types.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2022 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// ManagedZoneParameters define the desired state of a ManagedZone
+type ManagedZoneParameters struct {
+
+	// Description: A mutable string of at most 1024 characters associated
+	// with this resource for the user's convenience. Has no effect on the
+	// managed zone's function. Defaults to 'Managed by Crossplane'
+	// +optional
+	Description *string `json:"description,omitempty"`
+
+	// DNSName: The DNS name of this managed zone, for instance "example.com.".
+	// +immutable
+	DNSName string `json:"dnsName"`
+
+	// Labels: User labels.
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// PrivateVisibilityConfig: For privately visible zones, the set of
+	// Virtual Private Cloud resources that the zone is visible from.
+	// +optional
+	PrivateVisibilityConfig *ManagedZonePrivateVisibilityConfig `json:"privateVisibilityConfig,omitempty"`
+
+	// Visibility: The zone's visibility: public zones are exposed to the
+	// Internet, while private zones are visible only to Virtual Private
+	// Cloud resources. Defaults to 'public`
+	//
+	// Possible values:
+	//   "public"
+	//   "private"
+	// +optional
+	// +immutable
+	// +kubebuilder:validation:Enum=public;private
+	Visibility *string `json:"visibility,omitempty"`
+
+	// TODO(danielinclouds): support CloudLoggingConfig parameters
+	// TODO(danielinclouds): support DnssecConfig parameters
+	// TODO(danielinclouds): support ForwardingConfig parameters
+	// TODO(danielinclouds): support NameServerSet parameters
+	// TODO(danielinclouds): support PeeringConfig parameters
+	// TODO(danielinclouds): support ReverseLookupConfig parameters
+	// TODO(danielinclouds): support ServiceDirectoryConfig parameters
+}
+
+// ManagedZonePrivateVisibilityConfig the set of Virtual Private Cloud resources
+// that the zone is visible from
+type ManagedZonePrivateVisibilityConfig struct {
+
+	// Networks: The list of VPC networks that can see this zone.
+	Networks []*ManagedZonePrivateVisibilityConfigNetwork `json:"networks"`
```

Comment:
I believe we should also put it as optional

```suggestion
	// Networks: The list of VPC networks that can see this zone.
	// +optional
	Networks []*ManagedZonePrivateVisibilityConfigNetwork `json:"networks,omitempty"`
```